lib/js/jsfetch.nim

```diff
@@ -0,0 +1,120 @@
+## - Fetch for the JavaScript target: https://developer.mozilla.org/docs/Web/API/Fetch_API
+from httpcore import HttpMethod
+export HttpMethod
+
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsfetch is designed to be used with the JavaScript backend.".}
+when defined(nodejs):
+  {.warning: "By design Fetch is defined on the web browser, but may not be on NodeJS.".}
```

Comment:
maybe future work but https://www.npmjs.com/package/node-fetch could solve this (and allow testing this module too, which would be nice)